uint8 action_type
---
bool action_completed
---
bool action_progress